mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -239,6 +250,48 @@ private boolean textContains(String value, MailboxMessage message) throws IOExce
         return isInMessage(value, new SequenceInputStream(textHeaders(message), bodyContent), true);
     }
 
+    private boolean atachmentsContain(String value, MailboxMessage message) throws IOException, MimeException {
+        List<MessageAttachment> attachments = message.getAttachments();
+        return isInAttachments(value, attachments);
+    }
+
+    private boolean isInAttachments(String value, List<MessageAttachment> attachments) {
+        return attachments.stream()
+            .map(this::extractText)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .filter(string -> string.contains(value))
+            .findAny()
+            .isPresent();
+    }
+
+    private Optional<String> extractText(MessageAttachment attachment) {
+        if (isPDF(attachment)) {
```

Comment:
why do we handle a special case for PDF ?

Reply:
:+1: We may want to rely on TextExtractor.

Reply:
In order to have different parser without bringing complex things.

Reply:
TextExtractor is not a complex thing. TikaTextExtractor is. TextExtractor is only the abstraction.

I don't see the use of this hard coded **PDFTextExtractor**. I would prefer relying in the general abstraction API for this (as stated in the ticket description BTW)

Reply:
Ok